staging/src/k8s.io/api/node/v1alpha1/types.go

```diff
@@ -114,3 +114,76 @@ type RuntimeClassList struct {
 	// Items is a list of schema objects.
 	Items []RuntimeClass `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +k8s:conversion-gen:explicit-from=net/url.Values
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeLogViewerOptions is the query options for a Node's logs REST call. The options specified here intersect unless
+// called out otherwise.
+type NodeLogViewerOptions struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// SinceTime is an RFC3339 timestamp from which to show logs.
+	// +optional
+	SinceTime *metav1.Time `json:"sinceTime,omitempty" protobuf:"bytes,1,opt,name=sinceTime"`
+
+	// UntilTime is an RFC3339 timestamp until which to show logs.
+	// +optional
+	UntilTime *metav1.Time `json:"untilTime,omitempty" protobuf:"bytes,2,opt,name=untilTime"`
+
+	// TailLines is used to retrieve the specified number of lines (not more than 100k) from the end of the log.
+	// Support for this is implementation specific and only available for Linux and Windows service logs. It will be
+	// ignored for the unsupported case.
+	// +optional
+	TailLines *int64 `json:"tailLines,omitempty" protobuf:"varint,3,opt,name=tailLines"`
+
+	// Path is used to retrieve the contents of a literal file relative to the node's logs directory.
+	// If both a path and a service heuristic are specified, the service heuristic is performed. If neither is specified
+	// all services will be returned.
+	// +optional
+	Path string `json:"path,omitempty" protobuf:"bytes,4,opt,name=path"`
+
+	// Format is used to display service logs in an alternate format.
+	// Support for this is implementation specific. Supported only for service logs on Linux. It will be ignored for
+	// the unsupported case. Valid values are: short-precise, json, short, short-unix, short-iso, short-iso-precise,
+	// and cat. It will be ignored for the unsupported case.
+	// TODO: Make this into an enum type
+	// +optional
+	Format string `json:"format,omitempty" protobuf:"bytes,5,opt,name=format"`
+
+	// Pattern filters log entries by the provided regex pattern. On Linux nodes, this pattern will be read as a
+	// PCRE2 regex, on Windows nodes it will be read as a PowerShell regex.
+	// On Linux, if the pattern is all lowercase, matching is case insensitive. Otherwise, matching is case sensitive.
+	// This can be overridden with the PatternCaseSensitive option, see below.
+	// Support for this is implementation specific and only available for Linux and Windows service logs. It will be
+	// ignored for the unsupported case.
+	// +optional
+	Pattern string `json:"pattern,omitempty" protobuf:"bytes,6,opt,name=pattern"`
+
+	// PatternCaseSensitive is used to force case sensitivity or insensitivity the pattern for service logs on Linux nodes.
+	// By default case sensitivity of the search is derived from the Pattern string, see above.
```

Comment:
I'm not a fan of deriving the default value for this explicit behavior switch from implicit content of the pattern